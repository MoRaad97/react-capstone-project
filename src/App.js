import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import Navbar from './Components/Navbar';
import Home from './Components/Home';
import { loadDataThunk } from './Redux/Data_Reducer';

function App() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(loadDataThunk());
  }, []);

  const [theme, setTheme] = useState('default');

  const handleTheme = () => {
    if (theme === 'default') {
      setTheme('white');
    } else {
      setTheme('default');
    }
  };

  return (
    <>
      <Navbar theme={theme} themeSwitch={handleTheme} />
      <Home theme={theme} />
    </>
  );
}

export default App;
