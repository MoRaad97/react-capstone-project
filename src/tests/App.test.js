import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/Store';
import App from '../App';

it('App Component snapshot test', () => {
  const AppRender = TestRenderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>,
    )
    .toJSON();
  expect(AppRender).toMatchSnapshot();
});
