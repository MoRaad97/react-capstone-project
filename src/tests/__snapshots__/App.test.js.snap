// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component snapshot test 1`] = `
Array [
  <nav
    className="navbar df-nav"
  >
    <div
      className="container"
    >
      <h1>
        Where in the world?
      </h1>
      <h2
        onClick={[Function]}
      >
        White Theme
      </h2>
    </div>
  </nav>,
  <section
    className="home home-df"
  >
    <div
      className="container"
    >
      <form
        action=""
      >
        <div
          className="SearchBar-container"
          onBlur={[Function]}
          onChange={[Function]}
        >
          <input
            placeholder="Search for a country"
            type="search"
          />
        </div>
        <div
          className="filter-container"
        >
          <select
            name="Regions"
            onChange={[Function]}
          >
            <option
              value="All"
            >
              Filter by Region
            </option>
            <option
              value="Africa"
            >
              Africa
            </option>
            <option
              value="Americas"
            >
              Americas
            </option>
            <option
              value="Asia"
            >
              Asia
            </option>
            <option
              value="Europe"
            >
              Europe
            </option>
            <option
              value="Oceania"
            >
              Oceania
            </option>
          </select>
        </div>
      </form>
      <button
        className="scroll-up"
        onClick={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
      >
        â†‘
      </button>
      <div
        className="cards-container"
      />
    </div>
  </section>,
]
`;
